src/Resources/contao/library/Contao/User.php

```diff
@@ -256,58 +260,24 @@ public function getData()
 	 */
 	public function authenticate()
 	{
-		// Check the cookie hash
-		if ($this->strHash != $this->getSessionHash($this->strCookie))
-		{
-			return false;
-		}
+		$container = System::getContainer();
 
-		$objSession = $this->Database->prepare("SELECT * FROM tl_session WHERE hash=?")
-									 ->execute($this->strHash);
+		/** @var AuthorizationCheckerInterface $authorizationChecker */
+		$authorizationChecker = $container->get('security.authorization_checker');
 
-		// Try to find the session in the database
-		if ($objSession->numRows < 1)
-		{
-			return false;
-		}
+		/** @var TokenStorageInterface $tokenStorage */
+		$tokenStorage = $container->get('security.token_storage');
 
-		$time = time();
-		$container = \System::getContainer();
-		$session = $container->get('session');
-
-		// Validate the session
-		if ($objSession->sessionID != $session->getId() || (!$container->getParameter('contao.security.disable_ip_check') && $objSession->ip != $this->strIp) || $objSession->hash != $this->strHash || ($objSession->tstamp + \Config::get('sessionTimeout')) < $time)
```

Comment:
Which is an BC break then and the whole stuff has to get postponed to 5.0 then... I do not like this.